packages/block-library/src/columns/edit.js

```diff
@@ -0,0 +1,130 @@
+ * The contents of the array should never change.
+ * The array should contain the name of each block that is allowed.
+ * In columns block, the only block we allow is 'core/column'.
+ *
+ * @constant
+ * @type {string[]}
+*/
+const ALLOWED_BLOCKS = [ 'core/column' ];
+
+export const ColumnsEdit = function( { attributes, setAttributes, className, childColumns, updateChildColumnsAlignment } ) {
+	const { columns, verticalAlignment } = attributes;
+
+	// Do any of the child column have a valign setting that is different to what is currently
+	// set on this parent Columns Block?
+	const childColumnHasOveride = !! childColumns.find( ( childColumn ) => childColumn.attributes.verticalAlignment !== verticalAlignment );
+
+	// If we have an overide then remove the setting on the parent so that the toolbar
+	// no longer displays an alignment. This is inline with behaviour of user land editing
+	// tools such as MS Word...etc
+	if ( childColumnHasOveride ) {
+		setAttributes( { verticalAlignment: null } );
+	}
+
+	const classes = classnames( className, `has-${ columns }-columns`, {
+		[ `are-vertically-aligned-${ verticalAlignment }` ]: verticalAlignment,
+	} );
+
+	const onChange = ( alignment ) => {
+		// Update self...
+		setAttributes( { verticalAlignment: alignment } );
+
+		// Update all the (immediate) child Column Blocks
+		updateChildColumnsAlignment( alignment );
+	};
+
+	return (
+		<Fragment>
+			<InspectorControls>
+				<PanelBody>
+					<RangeControl
+						label={ __( 'Columns' ) }
+						value={ columns }
+						onChange={ ( nextColumns ) => {
+							setAttributes( {
+								columns: nextColumns,
+							} );
+						} }
+						min={ 2 }
+						max={ 6 }
+					/>
+				</PanelBody>
+			</InspectorControls>
+			<BlockControls>
+				<BlockVerticalAlignmentToolbar
+					onChange={ onChange }
+					value={ verticalAlignment }
+				/>
+			</BlockControls>
+			<div className={ classes }>
+				<InnerBlocks
+					template={ getColumnsTemplate( columns ) }
+					templateLock="all"
+					allowedBlocks={ ALLOWED_BLOCKS } />
+			</div>
+		</Fragment>
+	);
+};
+
+export default compose(
+	/**
+	 * Selects the child column Blocks for this parent Column
+	 */
+	withSelect( ( select, { clientId } ) => {
+		const { getBlocksByClientId } = select( 'core/editor' );
+
+		return {
+			childColumns: getBlocksByClientId( clientId )[ 0 ].innerBlocks,
+		};
+	} ),
+	withDispatch( ( dispatch, { childColumns } ) => {
+		return {
+			/**
+			 * Update all child column Blocks with a new
+			 * vertical alignment setting based on whatever
+			 * alignment is passed in. This allows change to parent
+			 * to overide anything set on a individual column basis
+			 *
+			 * @param  {string} alignment the vertical alignment setting
+			 */
+			updateChildColumnsAlignment( alignment ) {
+				childColumns.forEach( ( childColumn ) => {
+					dispatch( 'core/editor' ).updateBlockAttributes( childColumn.clientId, {
```

Comment:
It'd be cool if we could do something like this instead of having to use a loop:
```
dispatch( 'core/editor' ).updateAttributesForBlocks( [ // block ids... ], { // ... attributes } );
```

Does the loop cause multiple undo states? Tested and it seems like the undo button doesn't work after setting an alignment 🤔.

It seems like the alignment attribute is unset on the columns block, but not the child column. I guess no undo states are created for this action.

Reply:
> It'd be cool if we could do something like this instead of having to use a loop:

I've got to confess I'm not entirely sure how to achieve this.

I can see the [`action`](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/actions.js#L57) and the [`reducer`](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/reducer.js#L430). 

A top level thought would be to add the hypothetical `updateAttributesForBlocks` action creator which itself dispatches the `updateBlockAttributes()` action creator for each BlockId + Attributes combo it receives. 

Any guidance would be appreciated.

> Tested and it seems like the undo button doesn't work after setting an alignment 

I wish I could shed light on this. Is this a Blocker?

Reply:
On a different note, shouldn't it reset vertical alignment for child blocks instead? It seems like this would be a good compromise:
- child blocks would always inherit from the parent when their value is not set
- when child block gets updated then all other child blocks would have the value explicitly set based on the parent, then parent would get reset value applied